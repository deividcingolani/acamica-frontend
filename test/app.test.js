import React from "react";
import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';

import App from "../pages/index.js";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (props = {}, state = null) => {
  const wrapper = shallow(<App {...props} />);
  if (state) wrapper.setState(state);
  return wrapper;
};


const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

describe("test without Error index", () => {
  it('There is a Home"', () => {
    const wrapper = setup();
    const appComponent = findByTestAttr(wrapper, "index");
    expect(appComponent.length).toBe(1);
  });

  it('There is a Layout"', () => {
    const wrapper = setup();
    const appComponent = findByTestAttr(wrapper, "layout");
    expect(appComponent.length).toBe(1);
  });

  it('there is a button of login"', () => {
    const wrapper = setup();
    const appComponent = findByTestAttr(wrapper, "button-login");
    expect(appComponent.length).toBe(1);
  });

});
