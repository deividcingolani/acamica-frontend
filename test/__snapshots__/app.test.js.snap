// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello, Sunshine!" 1`] = `
<div
  className="jsx-3170664004"
>
  <div
    className="jsx-473655496 root"
  >
    <header
      className="jsx-473655496 headerApp"
    >
      <a
        className="jsx-473655496"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
      <li
        className="jsx-473655496 list"
      >
        <a
          className="jsx-473655496 login"
          onClick={[Function]}
        >
          Login / Register
        </a>
      </li>
    </header>
    <h1
      className="jsx-473655496"
    />
    <p
      className="jsx-3170664004"
    >
      This is an application for Students of Acamica
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Login/Register
    </button>
    <footer
      className="jsx-473655496"
    >
      Acamica Â© 
      2020
    </footer>
  </div>
</div>
`;
